src/server/ApiDispatchersInternal.cpp

```diff
@@ -90,11 +90,14 @@ using Microsoft::Console::Interactivity::ServiceLocator;
             RETURN_IF_NTSTATUS_FAILED((NtPrivApi::s_GetProcessParentId(&ProcessId)));
             ProcessHandle = gci.ProcessHandleList.FindProcessInList(ProcessId);
             RETURN_HR_IF_NULL(E_INVALIDARG, ProcessHandle);
-            RETURN_IF_FAILED(gci.ProcessHandleList.AllocProcessData(a->ProcessGroupId,
-                                                                    0,
-                                                                    a->ProcessGroupId,
-                                                                    ProcessHandle,
-                                                                    nullptr));
+
+            const auto hr = gci.ProcessHandleList.AllocProcessData(a->ProcessGroupId, 0, a->ProcessGroupId, nullptr);
+            // AllocProcessData returns E_FAIL if the given process ID already exists.
+            // In that case we don't really need to return an error and can just carry on.
+            if (FAILED(hr) && hr != E_FAIL)
+            {
+                RETURN_HR(hr);
+            }

```

Comment:
Now that `AllocProcessData` consistently returns `E_FAIL` if the process ID already exists, we need this bit of extra code here. TBH I don't quite understand why this adds a "ProcessGroupId" as a process ID to the process list here...